sympy/matrices/expressions/tests/test_matpow.py

```diff
@@ -70,7 +70,7 @@ def test_doit_square_MatrixSymbol_symsize():
 
 def test_doit_with_MatrixBase():
     X = ImmutableMatrix([[1, 2], [3, 4]])
-    assert MatPow(X, 0).doit() == ImmutableMatrix(Identity(2))
+    assert MatPow(X, 0).doit().as_explicit() == ImmutableMatrix(Identity(2))
```

Comment:
Why is the test changed? You shouldn't change a test unless it was testing against an incorrect solution.

You're new code should not affect existing tests (unless the tests were wrong), it should only need new tests added.